File: 283_move_zeros.cpp
class Solution {
public:
    int count=0;
    void moveZeroes(vector<int>& nums) {
        vector<int> arr;
        int count=0;
        for (int i=0 ;i<nums.size();i++){
            if(nums[i]!=0){
                nums[count++]=nums[i];
            }
        }
        for (int i=count;i<nums.size();i++){
            nums[i]=0;
        }
    }
};

File: Arrays/283MoveZeros.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {
    let countnonzero=0;
    for(let i=0;i<nums.length;i++){
        if(nums[i]!==0){
            // swaping sing index
            [nums[i],nums[countnonzero]]=[nums[countnonzero],nums[i]];
            countnonzero++;
        }
    }  
};




